Tasks/CopyFilesOverSSHV0/utils.ts

```diff
@@ -28,12 +28,15 @@ export function pathIsUNC(path: string): boolean {
  * @returns {string} OS specific command to clean target folder on the remote machine
  * @param {string} targetFolder path to target folder
  * @param {boolean} forWindows return command for Windows CMD
+ * @param {boolean} cleanHiddenFiles clean hedden files in target folder
  */
-export function getCleanTargetFolderCmd(targetFolder: string, forWindows: boolean): string {
+export function getCleanTargetFolderCmd(targetFolder: string, forWindows: boolean, cleanHiddenFiles: boolean = false ): string {
     if (forWindows) {
+        const cleanFilesInTarget = cleanHiddenFiles ? `del /q /A:H "${targetFolder}\\*"` : `del /q "${targetFolder}\\*"`
         // delete all files in specified folder and then delete all nested folders
-        return `del /q "${targetFolder}\\*" && FOR /D %p IN ("${targetFolder}\\*.*") DO rmdir "%p" /s /q`;
+        return `${cleanFilesInTarget} && FOR /D %p IN ("${targetFolder}\\*.*") DO rmdir "%p" /s /q`;
     } else {
-        return `sh -c "rm -rf '${targetFolder}'/*"`;
+        const cleanFilesInTarget = cleanHiddenFiles ? `sh -c "rm -rf '${targetFolder}'/{,.[!.],..?}*"` : `sh -c "rm -rf '${targetFolder}'/*"`;
```

Comment:
Could you please describe what is the purpose of '${targetFolder}'/{,.[!.],..?}*' here?
Is there some some linux doc/man page describing it as a preferable way to remove hidden files on linux? Should there be files with names starting with dot only?

Reply:
This pattern is required to delete only files that is starting with a dot or regular files. The problem with patterns in form like .* is that it will try to delete ".." and "." links from the directory, which is included in each Linux directory. Attempt to delete these links will produce warnings that could confuse users. Here is [an article](https://unix.stackexchange.com/questions/77127/rm-rf-all-files-and-all-hidden-files-without-error/77313#77313) related to this problem.

Reply:
Could you please describe it in the code comments - which patterns excluded, what is the purpose (short description of a problem) and link to this source?